tests/framework/cli/test_starters.py

```diff
@@ -130,61 +118,68 @@ def test_custom_project_name_with_hyphen(self, fake_kedro_cli):
-        result = CliRunner().invoke(
-            fake_kedro_cli,
-            ["new"],
-            input=_make_cli_prompt_input(repo_name="my-repo"),
-        )
-        _assert_template_ok(
-            result,
-            project_name="New Kedro Project",
-            repo_name="my-repo",
-            python_package="new_kedro_project",
-        )
+    def test_no_prompts(self, fake_kedro_cli):
+        shutil.copytree(TEMPLATE_PATH, "template")
+        (Path("template") / "prompts.yml").unlink()
+        result = CliRunner().invoke(fake_kedro_cli, ["new", "--starter", "template"])
+        _assert_template_ok(result)
+
+    def test_empty_prompts(self, fake_kedro_cli):
+        shutil.copytree(TEMPLATE_PATH, "template")
+        _write_yaml(Path("template") / "prompts.yml", {})
+        result = CliRunner().invoke(fake_kedro_cli, ["new", "--starter", "template"])
+        _assert_template_ok(result)
 
-    def test_custom_python_package(self, fake_kedro_cli):
+    def test_custom_prompt_valid_input(self, fake_kedro_cli):
+        shutil.copytree(TEMPLATE_PATH, "template")
+        _write_yaml(
+            Path("template") / "prompts.yml",
+            {
+                "project_name": {"title": "Project Name"},
+                "custom_value": {
+                    "title": "Custom Value",
+                    "regex_validator": "^\\w+(-*\\w+)*$",
+                },
+            },
+        )
+        custom_input = "\n".join(["my-project", "My Project"])
         result = CliRunner().invoke(
             fake_kedro_cli,
-            ["new"],
-            input=_make_cli_prompt_input(python_package="my_package"),
+            ["new", "--starter", "template"],
+            input=custom_input,
         )
         _assert_template_ok(
             result,
-            project_name="New Kedro Project",
-            repo_name="new-kedro-project",
-            python_package="my_package",
+            project_name="My Project",
+            repo_name="my-project",
+            python_package="my_project",
         )
 
-    def test_custom_all(self, fake_kedro_cli):
+    def test_custom_prompt_for_essential_variable(self, fake_kedro_cli):
+        shutil.copytree(TEMPLATE_PATH, "template")
+        _write_yaml(
+            Path("template") / "prompts.yml",
+            {
+                "project_name": {"title": "Project Name"},
+                "repo_name": {
+                    "title": "Custom Repo Name",
+                    "regex_validator": "^[a-zA-Z_]\\w{1,}$",
+                },
+            },
+        )
+        custom_input = "\n".join(["My Project", "my_custom_repo"])
         result = CliRunner().invoke(
             fake_kedro_cli,
-            ["new"],
-            input=_make_cli_prompt_input(
-                project_name="My Project",
-                repo_name="my-repo",
-                python_package="my_package",
-            ),
+            ["new", "--starter", "template"],
```

Comment:
is the `--starter` argument necessary? The old test is just generating project without any nodes.